### DISCUSSION TITLE
How to serialize *lua.LState?

### DISCUSSION THREAD

**Discussion by sledro** [yuin/gopher-lua#386] [Q&A]
Hey Guys,

I'm trying to serialize `*lua.LState` using the gob encoder so I can store the state in leveldb

I'm getting the following error: `runtime: goroutine stack exceeds 1000000000-byte limit`

How can I serialize`*lua.LState` ?

**Comment by assemblaj** [comment]
@Sledro did you ever find a solution? I'm from the [IKEMEN-Go](https://github.com/ikemen-engine/Ikemen-GO) project, we use Lua for some scripting, and we use gopher-lua as our VM. We are in the process of implementing rollback netcode, so we (ideally) would like to be able to save and return `*lua.LState` into a prior state.  When trying to create my own serialization functions I ran into the same issue with regards to the goroutine stack exceeding its limit.  I presume an even bigger issue is that a lot of the core data structures that might need to be deep copied as well are in fact private.  So outside of actually forking the library and rolling your own serializer I'm not sure what I'm to do here. @yuin do you have any suggestions?
